agent/api/ecsclient/client.go

```diff
@@ -123,21 +123,31 @@ func (client *APIECSClient) RegisterContainerInstance(containerInstanceArn strin
 
 func (client *APIECSClient) registerContainerInstance(clusterRef string, containerInstanceArn string, attributes []string) (string, error) {
 	registerRequest := ecs.RegisterContainerInstanceInput{Cluster: &clusterRef}
+	var registrationAttributes []*ecs.Attribute
 	if containerInstanceArn != "" {
+		// We are re-connecting a previously registered instance, restored from snapshot.
 		registerRequest.ContainerInstanceArn = &containerInstanceArn
+	} else {
+		// This is a new instance, not previously registered.
+		// Custom attribute registration only happens on initial instance registration.
```

Comment:
Custom attributes are ignored by the API on agent re-registration, and it's possible that a previously registered attribute has been deleted by an explicit API call. So we have no way of knowing what to expect for custom attributes in the response to `RegisterContainerInstance` when re-registering.